gallery_dl/extractor/postmill.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for Postmill instances"""
+
+import re
+import urllib.parse
+from .common import BaseExtractor, Message
+from .. import text, exception
+
+
+class PostmillExtractor(BaseExtractor):
+    """Base class for Postmill extractors"""
+    basecategory = "postmill"
+    directory_fmt = ("{category}", "{instance}", "{forum}")
+    filename_fmt = "{id}_{title[:220]}.{extension}"
+    archive_fmt = "{filename}"
+
+    def _init(self):
+        self.instance = self.root.partition("://")[2]
+        self.save_link_post_body = self.config("save-link-post-body", False)
+        self._search_canonical_url = re.compile(r"/f/([\w\d_]+)/(\d+)/").search
+        self._search_image_tag = re.compile(
+            r'<a href="[^"]+"\n +class="submission__image-link"').search
+
+    def items(self):
+        for post_url in self.post_urls():
+            response = self.request(post_url)
+            extr = text.extract_from(response.text)
+
+            title = text.unescape(extr(
+                '<meta property="og:title" content="', '">'))
+            date = text.parse_datetime(extr(
+                '<meta property="og:article:published_time" content="', '">'))
+            username = extr(
+                '<meta property="og:article:author" content="', '">')
+            post_canonical_url = text.unescape(extr(
+                '<link rel="canonical" href="', '">'))
+
+            url = text.unescape(extr(
+                '<h1 class="submission__title unheaderize inline"><a href="',
+                '"'))
+            body = extr(
+                '<div class="submission__body break-text text-flow">',
+                '</div>')
+
+            match = self._search_canonical_url(post_canonical_url)
+            forum = match.group(1)
+            id = int(match.group(2))
+
+            is_text_post = url.startswith("/")
+            is_image_post = self._search_image_tag(response.text) is not None
+            data = {
+                "title": title,
+                "date": date,
+                "username": username,
+                "forum": forum,
+                "id": id,
+                "flair": [text.unescape(i) for i in text.extract_iter(
+                    response.text, '<span class="flair__label">', '</span>')],
```

Comment:
Do not access `response.text` more than once (it's quite computationally expensive).
Store it in a `page` variable (or give it a different name, but I usually call it `page`) and use that.